diff -Naur freeswitch-1.10.1/src/mod/applications/mod_avmd/mod_avmd.c freeswitch-1.10.1-patch/src/mod/applications/mod_avmd/mod_avmd.c
--- freeswitch-1.10.1/src/mod/applications/mod_avmd/mod_avmd.c 2019-01-25 00:17:59.000000000 +0000
+++ freeswitch-1.10.1-patch/src/mod/applications/mod_avmd/mod_avmd.c 2019-03-21 21:28:21.375188575 +0000
@@ -1402,7 +1402,8 @@
     switch_channel_t    *channel = NULL;
     avmd_session_t      *avmd_session = NULL;
     switch_core_media_flag_t flags = 0;
-	const char *direction = "NO DIRECTION";
+    const char *direction = "NO DIRECTION";
+    uint8_t report = 0;

     if (session == NULL) {
         switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "BUGGG. FreeSWITCH session is NULL! Please report to developers\n");
@@ -1447,6 +1448,8 @@
             goto end;
     }

+    report = avmd_session->settings.report_status;
+
     status = init_avmd_session_data(avmd_session, session, avmd_globals.mutex);
     if (status != SWITCH_STATUS_SUCCESS) {
         switch (status) {
@@ -1508,7 +1511,9 @@
     status = switch_core_media_bug_add(session, "avmd", NULL, avmd_callback, avmd_session, 0, flags, &bug); /* Add a media bug that allows me to intercept the audio stream */
     if (status != SWITCH_STATUS_SUCCESS) { /* If adding a media bug fails exit */
         switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "Failed to add media bug!\n");
-        goto end_unlock;
+        switch_mutex_unlock(avmd_session->mutex);
+        avmd_session_close(avmd_session);
+        goto end;
     }

     switch_mutex_lock(avmd_globals.mutex);
@@ -1526,7 +1531,7 @@

 end:
     if (status != SWITCH_STATUS_SUCCESS) {
-        if (avmd_session == NULL || avmd_session->settings.report_status == 1) {
+        if (avmd_session == NULL || report) {
             switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "Avmd on channel [%s] NOT started\n", switch_channel_get_name(channel));
         }
     }
