diff -Naur freeswitch-1.8.5/configure.ac freeswitch-1.8.5-patch/configure.ac
--- freeswitch-1.8.5/configure.ac	2019-01-25 00:17:59.000000000 +0000
+++ freeswitch-1.8.5-patch/configure.ac	2019-03-21 21:28:21.375188575 +0000
@@ -414,15 +414,10 @@
 elif test "x${ax_cv_c_compiler_vendor}" = "xclang" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC -ffast-math)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC -ffast-math)
-    APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
 elif test "x${ax_cv_c_compiler_vendor}" = "xgnu" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC -ffast-math)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC -ffast-math)
     AC_SUBST([AM_MOD_AVMD_CXXFLAGS], [-std=gnu99])      # FS-8809, needed for MAP_POPULATE
-    if test "$ac_cv_gcc_supports_w_no_unused_result" = yes; then
-      APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
-      APR_ADDTO(SWITCH_AM_CFLAGS, -Wno-unused-result)
-    fi
     if test "$ac_cv_gcc_supports_w_no_misleading_indentation" = yes; then
       APR_ADDTO(SWITCH_AM_CFLAGS, -Wno-misleading-indentation)
     fi
